### ISSUE TITLE
Huge performance drop in hdStorm with high poly count meshes

### ISSUE THREAD

**Issue by pberto** [PixarAnimationStudios/USD#1249]
### Description of Issue

There is a serious draw performance issue with hdStorm (tested both on `20.05` and on latest `dev`) which produces sub-optimal FPS rates when rendering meshes with "large" poly counts. It can be easily verified using hdStorm on usdview. 
### Steps to Reproduce

I am including here a super simple repro USD file which you can use to verify this at your end: it's a 17M poly mesh quads plane which will drop the FPS performance in hdStorm to very poor results. Simply run usdview and load the file and try to tumble the view while looking at FPS.

File: https://www.dropbox.com/s/8gzen058x1xegfj/plane17M.usd?dl=1

This scene performs at 60fps with a standard vfx-level graphics card on other applications with traditional OpenGL viewports (e.g. Maya / Houdini...)

This has been verified on several hardware and O/S in our office, just to give you an example reference:

- Quadro P2000 5GB / Geforce GTX 1660 Super... etc
- Intel xeon e2136 (6core/12threads)
- Linux EL7

> NOTE: We also went back testing on an older `19.05` version and performance is better (though still not 60fps). We are pretty sure things were even better earlier (2017-ish) also please note this thread on your from that corroborates it: https://groups.google.com/d/msg/usd-interest/U0xkvvjpTF8/ok-V68k5CQAJ

### System Information (OS, Hardware)

Any.

### Package Versions

Latest `dev` or `20.05` 

### Build Flags

Built as `release`.

**Comment by c64kernal** [comment]
Thanks for the report. We have hundreds (and constantly growing number) of performance tests that we run against every version of USD/Hydra/Storm that we put out (dating before we open sourced it). We haven't noticed a drastic regression as described. Though, thank you for the repro case, we will look into it when we get a chance -- it's possible that we may not have that case covered.

**Comment by pberto** [comment]
One important thing to mention is to run this test **not** on a ultra high end graphics card which could be masking the actual issue. A mid range gfx card like the one mentioned above is able to preform at 60fps on a traditional GL viewport with such asset, but produces poor results with hdSt.

**Comment by jtran56** [comment]
Filed as internal issue #USD-6169.

**Comment by pberto** [comment]
Does this "filing" mean you managed to reproduce it? Could you offer a temporary workaround ad interim?

**Comment by c64kernal** [comment]
No, unfortunately we still don't have any new information @pberto -- we'll post here if we do.

**Comment by pberto** [comment]
Hello, any update about this? Could anyone load the asset provided above in `usdview` on a machine that uses a basic gpu?

**Comment by pberto** [comment]
Sorry to bother you guys, but could someone try the asset I uploaded on a basic gpu on your side? I’d like to get a confirmation.

**Comment by c64kernal** [comment]
Hi @pberto -- we've been looking at this case, and haven't seen any degradation in performance between our latest dev and very early builds. Granted, the performance isn't great, however we have no reason to believe that there's a regression here. Also, apologies, but our ability to test on basic GPUs is extremely limited, we don't have the resources to manage a true GPU test matrix so we concentrate on the higher end ones used in our productions. We also don't have the resources to optimize performance for the lower end GPUs, however, we'd appreciate any help, and if someone has a bug fix that improves performance on lower end GPUs without affecting performance on the higher end ones, we certainly would be happy to look at it.

**Comment by pberto** [comment]
> Granted, the performance isn't great, however we have no reason to believe that there's a regression here

@c64kernal Hi George, thanks for the quick reply. This is again probably because the performance regression is masked by the fact you use super good hardware and it can't be noticed. When we have some time we will make a couple of recordings and maybe profiling with nsight for you so you can see a regression on a lower end GPU such as with a QuadroP2000.

Out of curiosity which is your baseline production GPU?

**Comment by c64kernal** [comment]
Our baseline GPUs are currently P6000s

**Comment by pberto** [comment]
Has this been somehow alleviated with 21.08?

**Comment by c64kernal** [comment]
Hi @pberto -- we haven't been concentrating on this, so if there is any fix, it would be as a coincidence. If anyone in the community has access to the hardware you're targeting and wants to provide a PR, we'd be happy to consult on any questions.

**Comment by pberto** [comment]
This is happening still, tested on MBP on metal with 24.08:

```
Warning: in Resize at line 605 of openusd/pxr/imaging/hdSt/vboMemoryManager.cpp -- Attempting to resize the BAR with 0x6373ce6 elements when the max number of elements in the buffer array is 0x5555555. Clamping BAR size to the latter.
```

- performance still very sluggish and there is a raw of polygons that is colored oddly:

<img width="434" alt="image" src="https://github.com/user-attachments/assets/89ebccef-ee1d-46e8-9bfd-d7d03dae0afb">

Will check linux.
